manifests/config.pp

```diff
@@ -49,7 +49,7 @@
     class { '::foreman::config::passenger': } -> anchor { 'foreman::config_end': }
 
     if $foreman::ipa_authentication {
-      if $::default_ipa_server == '' or $::default_ipa_realm == '' {
+      if $::default_ipa_server == undef or $::default_ipa_server == '' or $::default_ipa_realm == undef or $::default_ipa_realm == '' {
```

Comment:
feels clunky, but OK.  My only idea is: `if !$::default_ipa_server or empty($::default_ipa_server) ...`